runtime/common/location.go

```diff
@@ -49,6 +51,33 @@ func LocationsMatch(first, second Location) bool {
 	return first.ID() == second.ID()
 }
 
+// LocationsInSameAccount returns true if both locations are nil,
+// if both locations are address locations when both locations have the same address,
+// or otherwise if their IDs are the same.
+//
+func LocationsInSameAccount(first, second Location) bool {
+	if first == nil && second == nil {
+		return true
+	}
+
+	if (first == nil && second != nil) || (first != nil && second == nil) {
+		return false
+	}
```

Comment:
i think we could simplify a bit:
```suggestion
	if first == nil {
		return second == nil
	}

	if second == nil {
		return first == nil   
		// or simply `return false`. But above is more verbose.
	}
```

Reply:
Good idea, simplified in ac4ae8b